app/models/signature.rb

```diff
@@ -65,4 +65,11 @@ def validated?
   def postal_district
     postcode.upcase[0..-4].match(/[A-Z]{1,2}[0-9]{1,2}[A-Z]?/).to_s
   end
+
+  def constituency
+    @constituency ||= ConstituencyApi::Client.constituencies(self.postcode).first
+  rescue ConstituencyApi::ConstituencyApiError
```

Comment:
using this kind of thing for example?
@logger = AuditLogger.new(Rails.root.join('log', logfilename), 'email reminders')